api/client/cli.go

```diff
@@ -160,6 +162,13 @@ func NewDockerCli(in io.ReadCloser, out, err io.Writer, clientFlags *cli.ClientF
 		}
 		cli.configFile = configFile
 
+		// Set the CLI's API version
+		if ver := os.Getenv("DOCKER_API_VERSION"); ver != "" {
+			api.Version = version.Version(ver)
+		} else if configFile.APIVersion != "" {
+			api.Version = version.Version(configFile.APIVersion)
```

Comment:
I think we should have some validation and restrictions;

- Validate that a properly formatted version (`[1-9].[0-9]+`) is passed (according to the tests `xxx` and `yyy` are accepted)
- The version should not be higher than the version that the client supports
- Possibly limit the minimum version (e.g. max 2 versions in the past); IIRC something like that was in the original use-case